indexer/indexer_tests/categories_test.cpp

```diff
@@ -119,6 +119,108 @@ UNIT_TEST(CategoriesHolder_Smoke)
   }
 }
 
+UNIT_TEST(CategoriesHolder_ReadableNameSmoke)
+{
+  classificator::Load();
+
+  auto const & categoriesHolder = GetDefaultCategories();
+  auto const & groupTranslations = categoriesHolder.GetGroupTranslations();
+
+  categoriesHolder.ForEachCategory([](CategoriesHolder::Category const & cat) {
+    for (auto const & synonym : cat.m_synonyms)
+    {
+      TEST_NOT_EQUAL(synonym.m_name[0], '^', ("symbol ^ is used incorrectly in categories.txt "
+                                              "and loaded to synonyms."));
+    }
+  });
+
+  for (auto const & group : groupTranslations)
+  {
+    for (auto const & translation : group.second)
+    {
+      TEST_NOT_EQUAL(translation.m_name[0], '^', ("symbol ^ is used incorrectly in categories.txt "
+                                                  "and loaded to group translations"));
+    }
+  }
+}
+
+UNIT_TEST(CategoriesHolder_ReadableName)
+{
+  char const kCategories[] =
+      "@shop\n"
+      "en:^Shop\n"
+      "ru:^Mагазин\n"
+      "\n"
+      "@meat\n"
+      "en:Beef|^Meat\n"
+      "ru:мясо\n"
+      "de:Schlachter\n"
+      "\n"
+      "@butcher\n"
+      "de:^Metzgerei\n"
+      "\n"
+      "shop|@shop\n"
+      "en:market\n"
+      "\n"
+      "shop-alcohol|@shop\n"
+      "en:Liquor Store|2^Alcostore\n"
+      "\n"
+      "shop-bakery|@shop\n"
+      "en:^buns\n"
+      "\n"
+      "shop-butcher|@meat|@butcher\n"
+      "en:2butcher\n"
+      "ru:3^Мясная лавка\n"
+      "de:Geschäft|2Laden\n"
+      "";
+
+  classificator::Load();
+  CategoriesHolder holder(make_unique<MemReader>(kCategories, sizeof(kCategories) - 1));
+
+  size_t count = 0;
+  holder.ForEachCategory([&count](CategoriesHolder::Category const & cat) {
+    if (count == 0)
```

Comment:
Not good to depend on a particular order in the CategoriesHolder. Better to test synonyms in accordance with category type, or name.